include/dpp/discordvoiceclient.h

```diff
@@ -100,6 +125,100 @@ struct DPP_EXPORT voice_out_packet {
 	uint64_t duration;
 };
 
+/**
+ * @brief Supported DAVE (Discord Audio Visual Encryption) protocol versions
+ */
+enum dave_version_t : uint8_t {
+	/**
+	 * @brief DAVE disabled (default for now)
+	 */
+	dave_version_none = 0,
+	/**
+	 * @brief DAVE enabled, version 1 (E2EE encryption on top of sodium)
+	 */
+	dave_version_1 = 1,
+};
+
+/**
+ * @brief Discord voice websocket opcode types
+ */
+enum voice_websocket_opcode_t : uint8_t {
+	voice_opcode_connection_identify = 0,
+	voice_opcode_connection_select_protocol = 1,
+	voice_opcode_connection_ready = 2,
+	voice_opcode_connection_heartbeat = 3,
+	voice_opcode_connection_description = 4,
+	voice_opcode_client_speaking = 5,
+	voice_opcode_connection_heartbeat_ack = 6,
+	voice_opcode_connection_resume = 7,
+	voice_opcode_connection_hello = 8,
+	voice_opcode_connection_resumed = 9,
+	voice_opcode_multiple_clients_connect = 11,
+	voice_opcode_client_connect = 12,
+	voice_opcode_client_disconnect = 13,
+	voice_opcode_media_sink = 15,
+	voice_client_flags = 18,
+	voice_client_platform = 20,
+	voice_client_dave_prepare_transition = 21,
+	voice_client_dave_execute_transition = 22,
+	voice_client_dave_transition_ready = 23,
+	voice_client_dave_prepare_epoch = 24,
+	voice_client_dave_mls_external_sender = 25,
+	voice_client_dave_mls_key_package = 26,
+	voice_client_dave_mls_proposals = 27,
+	voice_client_dave_mls_commit_message = 28,
+	voice_client_dave_announce_commit_transaction = 29,
+	voice_client_dave_mls_welcome = 30,
+	voice_client_dave_mls_invalid_commit_welcome = 31,
+};
+
+/**
+ * @brief DAVE E2EE Binary frame header
+ */
+#pragma pack(push, 1)
+struct dave_binary_header_t {
+	/**
+	 * @brief Sequence number
+	 */
+	uint16_t seq;
```

Comment:
These are uninitialized, is this intended?

Reply:
yes, they are initialised by a memcpy into the structure from the websocket process_frame function